dlt/common/data_writers/buffered.py

```diff
@@ -117,8 +117,11 @@ def write_data_item(self, item: TDataItems, columns: TTableSchemaColumns) -> int
             else:
                 new_rows_count = 1
         self._buffered_items_count += new_rows_count
-        # flush if max buffer exceeded
-        if self._buffered_items_count >= self.buffer_max_items:
+        # flush if max buffer exceeded, the second path of the expression prevents empty data frames to pile up in the buffer
+        if (
+            self._buffered_items_count >= self.buffer_max_items
+            or len(self._buffered_items) >= self.buffer_max_items
+        ):
             self._flush_items()
```

Comment:
Could we eliminate the comment by refactoring it to a method?
```suggestion
        self.flush_if_max_buffer_exceeded()
```

Reply:
Similarly, `_update_row_count(item)` might be neat.